src/main/webapp/pagination.js

```diff
@@ -0,0 +1,235 @@
+let currentPage = 1;
+let next = true;
+let recordsPerPage = 3;
+const noOfPages = 5;
+const extraPageInFrontAndBack = Math.floor(noOfPages/2);
+const LOGS = "logs" ;
+const ERRORS = "errors";
+let fileLength = Number.MAX_VALUE;
+let noOfRecordsOnLastPage = recordsPerPage;
+let lastPage = Number.MAX_VALUE;
+let data = new Array();
+
+// decrement by 1 on pressing previous button
+function prevPage() {
+    currentPage--;
+    next = false;
+    changePage(currentPage);
+}
+
+// increment by 1 on pressing next button
+function nextPage() {
+    currentPage++;
+    next = true;
+    changePage(currentPage);
+}
+
+// change content of page 
+async function changePage(page) {
+    const logs = document.getElementById("logs").getAttribute("aria-selected");
+    const searchString = document.getElementById("searchBar").value;
+    let fileName = document.getElementById("fileName").value;
+    const fileType = logs == "true" ? LOGS : ERRORS ;
+
+    currentPage = page;
+    if(page == 1 && fileType == ERRORS){
+        fileLength = await getCount(fileName+"error");
+    }
+    const fetchedPage = getPageToBeFetched(); 
+    if(currentPage != 1 ) {
+        display();
+    }
+    if(fetchedPage != -1) {
+       
+        const params = new URLSearchParams();
+        const startAndSize = getPageStartAndSize(fileType, fetchedPage);
```

Comment:
make it json.
`const pageSpecs = getPageSpecs(fileType, fetchedPage);`
size will be pageSpecs.size and start will be pageSpecs.start